File: Algorithms/Implementations/caesar_cipher.py
def crypt_string(string, characters_to_move):
    new_string = ""
    for letter in string:
        letter_ord = ord(letter)
        new_letter_ord = letter_ord
        if letter_ord >= 65 and letter_ord <=122 and (letter_ord <=90 or letter_ord >= 97):
            new_letter_ord = letter_ord + characters_to_move

            while (letter_ord <=90 and new_letter_ord > 90) or (letter_ord <=122 and new_letter_ord > 122):
                new_letter_ord -= 26
        new_string += chr(new_letter_ord)
    return new_string

if __name__ == "__main__":
    length_of_string = int(raw_input().strip())
    string = raw_input().strip()
    characters_to_move = int(raw_input().strip())

    print crypt_string(string, characters_to_move)

File: Algorithms/Implementations/caesar_cipher.c
#include <stdio.h>

void move_character(char * letter, int characters_to_move)
{
    int letter_ord = *letter;
    int new_letter_ord = letter_ord;
    if ((letter_ord >=65 && letter_ord <= 90) || (letter_ord >= 97 && letter_ord <= 122))
    {
        new_letter_ord += characters_to_move;
    }
    while ((letter_ord <=90 && new_letter_ord >90) || (letter_ord <= 122 && new_letter_ord > 122))
    {
        new_letter_ord -= 26;
    }
    char new_letter = new_letter_ord;
    *letter = new_letter;
}

char * crypt_string(int length_of_string, char * string, int characters_to_move)
{
    for(int i = 0; i < length_of_string; i++)
    {
        move_character(&string[i], characters_to_move);
    }
    return string;
}

int main()
{
    int length_of_string;
    scanf("%d", &length_of_string);

    char string[length_of_string];
    scanf("%s", string);

    int characters_to_move;
    scanf("%d", &characters_to_move);

    printf("%s\n", crypt_string(length_of_string, string, characters_to_move));
}
